### PR TITLE
[KYUUBI #2436] Add AlterTableRecoverPartitionsCommand for Spark Sql Authz PrivilegesBuilder

### LINKED ISSUE #2436: [Bug] AlterTableRecoverPartitionsCommand Should be Considered in Spark Sql Authz PrivilegesBuilder
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

Sql `MSCK REPAIR TABLE table_name`, in spark 3.2 or greater, will produce a `RepairTableCommand` which has been considered in the PrivilegesBuilder, but in spark 3.1 or below, it will produce a `AlterTableRecoverPartitionsCommand` which has not been considered in the PrivilegesBuilder.

### Affects Version(s)

master

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by packyan**
### _Why are the changes needed?_

To close #2436 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilder.scala [hunk 1]
```diff
@@ -206,6 +206,10 @@ object PrivilegesBuilder {
         outputObjs += tablePrivileges(oldTable, actionType = PrivilegeObjectActionType.DELETE)
         outputObjs += tablePrivileges(newTable)
 
+      case "AlterTableRecoverPartitionsCommand" =>
```
Comment (ulysses-you) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilder.scala:209]: can we add some comments like: it is for Spark 2.4 ?

**Review by yaooqinn — APPROVED** [review]

**Comment by ulysses-you** [comment]
seems the style failed,. please run `./dev/reformat` thank you

**Comment by packyan** [comment]
> seems the style failed,. please run `./dev/reformat` thank you
ok, done.

**Review by ulysses-you — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**